projects/vanilla/src/app/configurators/metadata-configurator.component.ts

```diff
@@ -7,43 +7,162 @@ import { ComponentConfig, ConfiguratorContext } from "@sinequa/ngx-ui-builder";
+        <input class="form-check-input" type="checkbox" id="showEntityExtract" [(ngModel)]="metadataToModify.showEntityExtract" (ngModelChangeDebounced)="configChanged()">
+        <label class="form-check-label" for="showEntityExtract">Show entity extract</label>
+      </div>
+      <div class="mb-2" *ngIf="isEntity">
+        <label class="form-label" for="entityExtractMaxLines">Max lines for entity extract</label>
+        <input type="number" class="form-control" id="entityExtractMaxLines" [(ngModel)]="metadataToModify.entityExtractMaxLines">
+      </div>
+    </ng-container>
   </div>
 </div>
     `
 })
 export class MetadataConfiguratorComponent implements OnChanges {
   @Input() context: ConfiguratorContext;
-  @Input() metadata: MetadataConfig[];
+  @Input() metadata: string[];
 
+  modifiedMetadata: string;
+  metadataString: string;
+  metadataToModify: MetadataConfig;
+  stringToModify?: string;
+  newString: string;
   isEntity: boolean;
 
-  constructor(public appService: AppService) {}
+  constructor(public appService: AppService) { }
 
-  ngOnChanges(): void {
-    this.isEntity = this.appService.isEntity(this.config.item);
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.context && this.metadataToModify) {
+      this.metadataToModify = this.config.metadataConfig.find(m => m.field === this.metadataToModify.field);
+    }
   }
 
   get config(): ComponentConfig {
     return this.context.config;
   }
 
-  configChanged(isItem?: boolean) {
-    if (isItem) {
-      this.isEntity = this.appService.isEntity(this.config.item);
-    }
+  get metadataSelection(): string[] {
+    const stringItems = this.config.metadataConfig?.filter(c => !c.field) || [];
+    return this.metadata.concat(stringItems);
+  }
+
+ /** Update the whole config */
+  configChanged() {
     this.context.configChanged();
   }
+
+  /** Selecting a metadata to update */
+  selectedMetadata() {
+    this.metadataToModify = this.config.metadataConfig.find(m => m.field === this.modifiedMetadata);
+    this.stringToModify = this.metadataToModify ? undefined : this.modifiedMetadata;
+    this.newString = String(this.stringToModify);
+    this.isEntity = this.metadataToModify && this.appService.isEntity(this.metadataToModify.field);
+  }
+
+  /** Insert a static string inside the metadata list */
+  insertString() {
+      if (this.metadataString.trim() !== "") {
+        this.config.metadataConfig.push(this.metadataString);
+        this.config.items.push(this.metadataString);
+        this.metadataString = '';
+        this.configChanged();
+      }
+  }
+
+  /** Update a static string */
+  changedString() {
+    if (this.newString) {
+      let metadataConfigValue = this.config.metadataConfig.find(m => m === this.stringToModify);
+      let itemsValue = this.config.items.find(m => m === this.stringToModify);
+
+      this.config.metadataConfig[this.config.metadataConfig.indexOf(metadataConfigValue)] = this.newString;
```

Comment:
`this.config.metadataConfig[this.config.metadataConfig.indexOf(metadataConfigValue)]` and `metadataConfigValue` are the same object

Reply:
I tried updating both of the values (doing like `if (metadataConfigValue) metadataConfigValue = this.newString;` since if I don't make the "if" I get a compilation error for the variable not being read), but it doesn't want to work